### ISSUE TITLE
Setup releases to publish to choco and/or winget package managers

### ISSUE THREAD

**Issue by ChristopherHaws** [microsoft/artifacts-credprovider#417]
Hey there!

I like to keep all my software up to date by using package managers such as `apt`, `choco`, `winget`, etc. I was wondering if it would be possible to setup an automated release to a package manager so that the Artifacts Credential Provider can be updated along with my other software rather than me having to subscribe to releases on github to learn about new versions?

Thanks!

**Comment by ShadowXVII** [comment]
I had mentioned something similar in #370 but they won't accept feature requests here.

If you would like to upvote/comment on the following it might help get some attention:
https://developercommunity.visualstudio.com/t/Add-installer-options-for-Azure-Artifact/10308779

**Comment by JohnSchmeichel** [comment]
We're no longer forwarding to the developer community, so having issues here is appropriate.

**Comment by ShadowXVII** [comment]
@JohnSchmeichel , great another overnight change :), so the two issues that were closed #370 and #379 that were closed can be added to this request.

**Comment by ShadowXVII** [comment]
Not stale.

**Comment by ShadowXVII** [comment]
🙄

**Comment by jeffreyrivor** [comment]
It seems like this should be possible now in winget now that Windows Package Manager manifests allow portable zip, although I can't tell at first glance if the zip must contain an installer or if the contents can just be extracted into a directory directly as the "installation."

**Comment by jeffreyrivor** [comment]
I just linked a PR in winget-pkgs to this issue take a crack at integrating into winget, but the standalone zip with lack of an installer (WiX/Nullsoft/etc.) seems to make it unable to be fully automatic with winget; the install location can't be preset to the NuGet plugin directory, or if this can be set in the manifest I couldn't tell what YAML property would do it.

**Comment by Dan-Albrecht** [comment]
@jeffreyrivor, is there an actual plan here? I'm setting up a new machine now and cannot use your official install script  `https://aka.ms/install-artifacts-credprovider.ps1`, because of the rate limit problems mentioned in multiple other won't fix issues (#244, #369, #380).

![image](https://github.com/microsoft/artifacts-credprovider/assets/7351388/d6325d6a-ab18-4660-afac-3f6484e3ec82)

Supporting winget would be perfect here.

**Comment by jeffreyrivor** [comment]
Unfortunately, winget doesn't support raw zip (e.g. no nested installer exe/msi/etc) in the structure that would be required to place the artifacts credprovider in the NuGet plugin location.
1. As noted in the [linked PR](https://github.com/microsoft/winget-pkgs/pull/132585), the `--location` argument would need to be manually specified to the NuGet plugin directory `%USERPROFILE%\.nuget\plugins`, there's no way in the winget YAML definition to default this value.
2. winget adds the executable extracted from the raw zip to PATH via a symlink.
3. Additionally, winget PRs are tested against an integration that attempts to ensure that raw zips that just extract an executable (and not launch the extracted executable as an installer) that are added to PATH are executable on that PATH; this test fails because the credprovider needs to load DLL dependencies that fail to handle symlinking properly. This is blocking the PR in its current state.

In summary, unless winget changes to allow a different default location (in this case, the NuGet plugin directory) and to allow disabling symlinking and adding to PATH for raw zipped exe extraction (NuGet doesn't need to use PATH to locate it anyhow), the credprovier will have to be packaged into an actual installer (using WiX/Nullsoft/Inno/etc) in order to be ingested into winget.

**Comment by kimsey0** [comment]
_Can_ we package it into an actual installer? If so, is there a preferred install system, for example one of those you mentioned (WiX/Nullsoft/Inno)? I guess the installer should just reproduce the behavior of the current install script closely.

**Comment by namra98** [comment]
Best solution would be to make MSI out of it and publish to winget? since all this script is doing is some installation?

**Comment by kimsey0** [comment]
[The script](https://github.com/microsoft/artifacts-credprovider/blob/8464eedc682e4e5f2f7c688df34ef1a8c74c03a4/helpers/installcredprovider.ps1) has a bit of logic to handle .NET versions, but other than that, it just copies a DLL into a specific folder.

**Comment by kimsey0** [comment]
And if we're to make an MSI, we need to decide on an MSI creation tool, like [WiX](https://wixtoolset.org/) or [NSIS](https://nsis.sourceforge.io/Main_Page) mentioned above.

**Comment by XVII** [comment]
Not actually stale.

**Comment by kimsey0** [comment]
This would still be very useful.

**Comment by PSFW** [comment]
This would be very useful. Its amazing that Microsoft is still putting out tools that are not included in WinGet. It's a disjointed user experience.

**Comment by kimsey0** [comment]
Bump.

**Comment by EraYaN** [comment]
This is not stale

**Comment by pierluc-mc** [comment]
Another option would be to release as a dotnet tool? `dotnet tool install --global Microsoft.NuGet.CredentialProvider`

**Comment by EraYaN** [comment]
That is probably the more "correct" option given that it is a tool for dotnet.

**Comment by kimsey0** [comment]
Not stale.
